install/infra/modules/eks/database.tf

```diff
@@ -0,0 +1,57 @@
+resource "random_password" "password" {
+  count = var.enable_external_database ? 1 : 0
+
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "aws_db_subnet_group" "gitpod_subnets" {
+  count = var.enable_external_database ? 1 : 0
+
+  name       = "db-sg-${var.cluster_name}"
+  subnet_ids = [module.vpc.public_subnets[2], module.vpc.public_subnets[3]]
+}
+
+resource "aws_security_group" "rdssg" {
+  count = var.enable_external_database ? 1 : 0
+
+  name   = "dh-sg-${var.cluster_name}"
+  vpc_id = module.vpc.vpc_id
+
+  egress {
```

Comment:
Per the [terraform aws_security_group documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group), it looks like this inline `egress` rule will be overwritten by the following `aws_security_group_rule`. If we drop this egress rule does the database still work?

Some further searching indicates that [inline rules are somewhat badly behaved](https://github.com/hashicorp/terraform-provider-aws/issues/1555), let's see if we can chip at this problem and keep all of the rule definitions in separate resources.

Reply:
You are absolutely right! this `egress` rule is basically unnecessary in this context. I have removed this.